/**:
  ros__parameters:
    use_sim_time: true

front_cloud_to_scan:
  ros__parameters:
    use_sim_time: True
    frame_id: front_camera_link
    pointcloud_topic: /front_camera/points
    scan_topic: /front_camera/scan
    frequency: 30
    x_max: 32.0
    x_min: 0.7
    y_max: 20.0
    y_min: -20.0
    z_max: 0.0
    z_min: -0.15
    angle_max: 0.7
    angle_min: -0.7
    rays_number: 200

path_mapping:
  ros__parameters:
    frequency: 30
    front_scan_topic: /front_camera/scan
    rear_scan_topic: /rear_camera/scan
    front_scan_position: [ 2.2, 0.0, 0.0 ]
    rear_scan_position: [ -2.2, 0.0, 0.0 ]
    map_topic: /map
    map_infiltration_radius: 2.5
    map_size: 200
    local_map_size: 40
    map_resolution: 0.1
    map_frame: map
    robot_base_frame: chassis
    odom_topic: /odom
    goal_topic: /goal_pose
    path_topic: /path
    path_collision_radius: 1.5
    goal_radius: 2.
    steering_value: 0.35
    path_discrete: 1.5
    path_base_frame: map
    grid_resolution: 0.1
    timeout: 0.2

path_controller:
  ros__parameters:
    frequency: 30
    path_topic: /path
    odom_topic: /odom
    cmd_topic: /cmd_vel
    p_steering_ratio: 3.0
    p_speed_ratio: 1.0
    average_speed: 1.5

waypoint_following:
  ros__parameters:
    frequency: 30
    odom_topic: /odom
    goal_topic: /goal_pose
    goal_radius: 5.0
    waypoints: [20.6453, -9.80267, 0., 24.3003, -31.9225, 0., 13.7962, -46.5684, 0., 20.6453, -9.80267, 1.]
    map_frame: map
    start_following: False # Начинать следование только после запуска и инициализации всех нод

/costmap/costmap:
  ros__parameters:
    debug_info: true
    use_sim_time: True
    global_frame: odom
    robot_base_frame: base_footprint
    update_frequency: 50.0
    publish_frequency: 50.0
    always_send_full_costmap: True
    
    width: 100
    height: 100
    origin_x: -50.0
    origin_y: -50.0
    resolution: 0.1
    
    rolling_window: true
    width_in_meters: 100
    height_in_meters: 100
    
    track_unknown_space: false
    
    transform_tolerance: 0.5
    obstacle_range: 25.0
    raytrace_range: 30.0
    
    plugins: ["obstacle_layer", "inflation_layer"]
    
    obstacle_layer:
      plugin: "nav2_costmap_2d::ObstacleLayer"
      enabled: True
      observation_sources: laser_scan
      laser_scan:
        topic: /front_camera/scan
        sensor_frame: "front_camera_link"
        reliability: reliable
        durability: volatile
        history: keep_last
        max_obstacle_height: 20.0
        min_obstacle_height: -10.0
        obstacle_max_range: 25.0
        obstacle_min_range: 0.0
        raytrace_max_range: 30.0
        raytrace_min_range: 0.0
        clearing: True
        marking: True
        data_type: "LaserScan"
        inf_is_valid: false

    inflation_layer:
      plugin: "nav2_costmap_2d::InflationLayer"
      enabled: True
      inflation_radius: 1.5
      cost_scaling_factor: 1.0
      inflate_unknown: true
      inflate_around_unknown: true